datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -577,6 +595,8 @@ struct BufferedBatch {
     /// The indices of buffered batch that failed the join filter.
     /// When dequeuing the buffered batch, we need to produce null joined rows for these indices.
     pub join_filter_failed_idxs: HashSet<u64>,
+    pub num_rows: usize,
```

Comment:
It would also help if we could document how to interpret `null_joined` -- specifically what do the indicies represent if both `batch` and `spill_file` are set? ? Or is a single buffered batch either in memory or spilled?

Reply:
documented newly added struct fields and null_joined